### DISCUSSION TITLE
Search entities menu showing multiple Certificate profiles under single SubCA

### DISCUSSION THREAD

**Discussion by merryo11** [Keyfactor/ejbca-ce#531] [General]
I have a CA under which I created 4 certificate profiles and created 4 RA roles so that each RA user with specific role can have access to its own certificate profile while doing the searching through UI but on advance search menu when I select the certificate profile option I can see all the 4 certificate profiles so my question is how can I lock it for RA to only see its allowed certificate profile and not others. 

Lastly, please also confirm how can I remove default profiles from the search options like ENDUSER,SUBCA,SERVER....

See screen shot below:

  
![MultipleProfiles-Under-OneCA](https://github.com/Keyfactor/ejbca-ce/assets/144104956/158e6cdb-dd33-4383-923c-07a44c60b2c8)

**Comment by merryo11** [comment]
any update on the community

**Reply by svenska-primekey** [reply]
Use different end entity profiles for each RA role You could go one step further and create each role in a namespace. If you use a namespace create a different cert profile for each end entity profile too.

**Reply by merryo11** [reply]
Thanks svenska, From the above screen shot two TMB certificate profiles comes under TMB entity while MMBL cert profile comes under MMBL entity but no luck so far. Secondly, I am not clear with the word namespace in this context.

**Reply by svenska-primekey** [reply]
Namespaces are explained here: https://doc.primekey.com/ejbca/ejbca-operations/ejbca-operations-guide/ca-operations-guide/roles-and-access-rules-operations/managing-role-namespaces

How many end entity profiles did you create?
